packages/matplotlib/src/html5_canvas_backend.py

```diff
@@ -382,11 +366,24 @@ def _draw_math_text(self, gc, x, y, s, prop, angle):
             self.ctx.restore()
 
     def draw_text(self, gc, x, y, s, prop, angle, ismath=False, mtext=None):
-        def _load_font_into_web(loaded_face):
-            document.fonts.add(loaded_face.result())
-            window.font_counter += 1
+        def _load_font_into_web(loaded_face, font_url):
+            fontface = loaded_face.result()
+            document.fonts.add(fontface)
+            self.fonts_loading.pop(font_url, None)
+
             self.fig.draw()
 
+            # For font load testing
+            if os.environ.get("TESTING_MATPLOTLIB"):
+                target_font = os.environ.get("TESTING_MATPLOTLIB_FONT")
+                if not target_font or target_font == fontface.family:
+                    try:
+                        from js import resolve
+
+                        resolve()
+                    except Exception as e:
+                        raise ValueError("unable to resolve") from e
+
```

Comment:
What about calling here `self._test_helper(fontface)` and then:
```py
def _test_helper(self):
	pass
```
then monkey patch this logic into `_test_helper` in the test suite?

Reply:
I'll try monkey-patching  `_load_font_into_web` directly.